app/src/main/java/org/sopt/hyundaicarddive/presentation/ui/detail/DetailScreen.kt

```diff
@@ -1,41 +1,242 @@
+import androidx.compose.foundation.background
 import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.aspectRatio
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.pager.HorizontalPager
+import androidx.compose.foundation.pager.PagerState
+import androidx.compose.foundation.pager.rememberPagerState
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.HorizontalDivider
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.sopt.hyundaicarddive.R
+import org.sopt.hyundaicarddive.core.component.TopBar
+import org.sopt.hyundaicarddive.presentation.model.DetailArticleModel
+import org.sopt.hyundaicarddive.presentation.model.DetailModel
+import org.sopt.hyundaicarddive.presentation.type.TopBarType
+import org.sopt.hyundaicarddive.presentation.ui.detail.component.DetailDescriptionSection
+import org.sopt.hyundaicarddive.presentation.ui.detail.component.DetailPager
+import org.sopt.hyundaicarddive.presentation.ui.detail.component.DetailTabBar
 import org.sopt.hyundaicarddive.ui.theme.HYUNDAICARDDIVETheme
+import org.sopt.hyundaicarddive.ui.theme.HYUNDAICARDDIVETheme.colors
+import org.sopt.hyundaicarddive.ui.theme.HYUNDAICARDDIVETheme.typography
 
 @Composable
 fun DetailRoute(
     padding: PaddingValues,
-    navigateToSpace: () -> Unit
+    navigateToSpace: () -> Unit,
+    viewModel: DetailViewModel = hiltViewModel()
 ) {
+    LaunchedEffect(Unit) {
+        viewModel.getDetail()
+    }
+    val detailModel by viewModel.detailModel.collectAsStateWithLifecycle()
+    val articleListSize by viewModel.articleListSize.collectAsStateWithLifecycle()
+
+    val pagerState = rememberPagerState(pageCount = { articleListSize })
```

Comment:
지금 articleListSize를 viewModel의 상태로 따로 관리하고 있는데, 이 값은 detailModel.articleList.size로부터 직접 유도될 수 있는 값이라 별도로 굳이 상태로까지 관리할 필요는 없는 것 같아요 ~~
```suggestion
val pagerState = rememberPagerState(pageCount = {
    viewModel.detailModel.value?.articleList?.size ?: 0
})

```

요런느낌으로다가 pagerState 생성 시점에 detailModel의 값을 기반으로 바로 전달해주면 상태를 관리하지 않고도 바로바로 사용할 수 있겠네요 !
이러면 viewmodel의 책임을 최소화하고, 불필요한 상태관리도 줄일 수 잇어욤

Reply:
해당 코드로 작성해봤는데 List가 비었다는 오류가 떠 아래와 같은 코드로 변경해보았습니다!
`val pagerState = rememberPagerState(pageCount = {
        detailModel?.articleList?.size ?: 0
    })`